STYLEGUIDE.md

```diff
@@ -0,0 +1,221 @@
+
+**Before** starting on any work, please check the [Project Board](https://github.com/Bannerlord-Modding/Documentation/projects/1) to ensure that nobody else is working on whatever you are planning on doing. If nobody is, [create an issue](https://github.com/Bannerlord-Modding/Documentation/issues) describing what you are creating / fixing / enhancing.
+
+All Pull Requests are expected to match this style guide.
+___
+
+## Base level Structure / Navigation
+
+Items in the root README.md should always be nested under wider category.
+
+Base level items are containers for more specific items and should be linked to the root README.me. Inidvidual files (like tutorials or classes) should not be linked to the root README.md, but should be detailed the lower level files. 
+
+```
+## [C# API Documentation](_csharp-api/)
+
+* [CampaignSystem](_csharp-api/campaignsystem/)
+* [Core](_csharp-api/core/)
+```
+All README files should only go two levels deep at *most* (but one may be more appropriate in many cases), any need to go deeper should be put into a README in the second level to avoid clutter in any single file.
+
+Each of the sections below would be the scope of a single readme:
+```
+ROOT README
+# Section (eg: api)
+ - packages
+```
+```
+# package
+ - classes
+```
+```
+# Class
+ - methods
+  - params
+```
+Each of these READMEs should only contain the table of contents of it's relevant data.
+___
+
+## C# Packages
+
+Each package directory should contain the following sections in its README:
+```md
+<Package overview>
+## Table of contents
+* [classes](_csharp-api/packagename/classes/)
+* [childclasses](_csharp-api/packagename/childclasses/)
+* ...
+## Classes
+* [classname](_csharp-api/packagename/classes/classname/)
+* ...
+## Child Classes
+* [classname](_csharp-api/packagename/childclasses/classname/)
+* ...
+## Template Classes
+* [classname](_csharp-api/packagename/templateclasses/classname/)
+* ...
+## Enums
+* [enumname](_csharp-api/packagename/enums/enumname/)
+* ...
+## Structs
+* [structname](_csharp-api/packagename/structs/structname/)
+* ...
+## Interfaces
+* [interfacename](_csharp-api/packagename/interfaces/interfacename/)
+* ...
+## Exception Classes
+* [exceptionname](_csharp-api/packagename/exceptions/exceptionname/)
+* ...
+```
+>Each of the sections (at both levels) should have a valid link to their respective info pages.
+
+The package information header should contain the following:
+```md
+Current Version (Stable branch): v0.0.0.0
+<Brief description of the package and its contents>
+```
+___
+
+## Classes (Base, Child, Templated, and Exceptions)
+
+Unlike the root and package level READMEs, the class README will contain more than just links to child folders. Each class will start with a **Class Overview Panel**:
+
+### [Parent Object Name](EXAMPLE)
```

Comment:
File should start with H1 header.
I suggest move that info to H2 after description.

Reply:
I'd argue its good to have quick access to the parrent class right next to the classname, but thats just me.

Reply:
Yeah, it's good. So that's why I suggested h2 after description.

But when you open a class readme - you, most likely want to read about this certain class first, not related.